rain-control system(modified after implementation of DayManager)

Kihara001 opened the pull request:
Implemented a rain particle system.
To integrate it with the dialogue system, use Raincontrol.Instance

kam-zhan-yue reviewed (changes requested):
Let's not make RainControl a static manager. All its doing is playing particles. Just put this logic inside of DayManager. So, get rid of RainControl class and put it in DayManager.

I don't wanna see any changes to scenes, only DayManager prefab.

Kihara001 commented:
Hi, I updated DayManager.cs and prefab as you said, but I pushed and gitignore adn DS.store file mistakenly.
I moved changes of gitignore file manually, but i don't know how to remove DS.store one.

Kihara001 commented:
I’ve increased the speed and frequency of the rain particles. However, I’m having trouble implementing collision between the rain particles and the ground. This is because the DayManager is part of a DontDestroyOnLoad scene, while the ground exists in the NoonlandEnvironment scene. They belong to different scenes, the particle system cannot interact with the ground's colliders.

kam-zhan-yue commented:
Yeah still not ready. The rain is not really visible. You need to make it more concentrated on the actual island. It's too sparse now
<img width="809" alt="Screenshot 2025-05-12 at 10 50 53 AM" src="https://github.com/user-attachments/assets/4e4e5152-7099-4944-b875-ec09faf85e2f" />

kam-zhan-yue reviewed (approved):
I jumped in and adjusted the rain particles as well as changing the material on the skybox. It's coming together, but could use a lot of polish in the next few months.
